### PR TITLE
TLS follow-up fixes

### PR DESCRIPTION
**Pull request by jrajahalme**
These commits address the rest of the review comments on #9486

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9807)

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

File: pkg/crypto/certificatemanager/certificate_manager.go [hunk 1]
```diff
@@ -87,9 +87,7 @@ func (m *Manager) GetTLSContext(ctx context.Context, tlsCtx *api.TLSContext, ns
 		// Give priority to local certificates
 		certPath := filepath.Join(m.rootPath, ns, name)
 		files, ioErr := ioutil.ReadDir(certPath)
-		if ioErr != nil {
-			err = fmt.Errorf("Certificates directory %s not found (%s)", certPath, ioErr)
-		} else {
+		if ioErr == nil {
```
Comment (aanm) [inline pkg/crypto/certificatemanager/certificate_manager.go:90]: How do we know if the directories didn't exist in the host in the first place?

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/crypto/certificatemanager/certificate_manager.go (hunk 1 above)
Comment (jrajahalme) [reply pkg/crypto/certificatemanager/certificate_manager.go:90]: The `err` set here was overwritten below, so in the current version we did not know it either. Also, the directory is not required to exist. Maybe we should require the existence of the directory if it is configured on the command line?

Comment (aanm) [reply pkg/crypto/certificatemanager/certificate_manager.go:90]: > The `err` set here was overwritten below, so in the current version we did not know it either. Also, the directory is not required to exist. Maybe we should require the existence of the directory if it is configured on the command line?

By "command line" you mean in the CLI or the k8s policies descriptors?

Comment (jrajahalme) [reply pkg/crypto/certificatemanager/certificate_manager.go:90]: I mean a cilium-agent command parameter (like `--debug`).

**Review by aanm — APPROVED** [review]

**Pull request merged**